server/expression.js

```diff
@@ -134,27 +134,43 @@ expression.query.list = function(expressionPath) {
  */
 expression.getTfaProfile_ = function(fileName, geneNames, conditionNames) {
   // TFA profile has the same format as an expression matrix.
-  var result = expression.readExpression_(fileName, geneNames, conditionNames);
+  // With empty geneNames and conditionNames input, returns the whole matrix.
+  var result = expression.readMatrix_(fileName, [], []);
```

Comment:
How do you differentiate selectAll from no selection? I think empty array means no selection, while null selection could instead mean all selection.